src/factories/node.ts

```diff
@@ -29,34 +32,54 @@ export async function nodeContainerFactory(node: RunGraphNode) {
 
   container.on('click', event => {
     event.stopPropagation()
-    selectNode(node)
+    selectNode(internalNode)
   })
 
+  if (!node.end_time) {
+    startTicking()
+  }
+
   emitter.on('nodeSelected', () => {
     const isCurrentlySelected = isSelected(node)
 
     if (isCurrentlySelected !== nodeIsSelected) {
       nodeIsSelected = isCurrentlySelected
-      renderNode(node)
+      renderNode(internalNode)
     }
   })
 
   async function render(node: RunGraphNode): Promise<BoundsContainer> {
+    internalNode = node
+
     const currentCacheKey = getNodeCacheKey(node)
 
     if (currentCacheKey === cacheKey) {
       return container
     }
 
+    cacheKey = currentCacheKey
+
     await renderNode(node)
 
-    if (!node.end_time) {
-      Ticker.shared.addOnce(() => render(node))
+    if (node.end_time) {
+      stopTicking()
     }
 
     return container
   }
 
+  function startTicking(): void {
+    application.ticker.add(tick)
+  }
+
+  function stopTicking(): void {
+    application.ticker.remove(tick)
+  }
+
+  function tick(): void {
+    render(node)
+  }
+
```

Comment:
Previous ticker implementation would just keep ticking because it would call render with the same value. So even after the node entered completed it would keep ticking. Nothing visible happened but waste of processing.